### PR TITLE
use super parameters

### LINKED ISSUE #631: use super parameters
use super parameters, once it is completely incorporated into the framework.

### PR DESCRIPTION
**Pull request by amanv8060**
Use super parameters

fixes #631 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -27,3 +27,4 @@ linter:
     - unawaited_futures
     - unnecessary_statements
     - unsafe_html
+    - use_super_parameters
```
Comment (amanv8060) [inline analysis_options.yaml:30]: should I add this line?

File: lib/demos/material/divider_demo.dart [hunk 2]
```diff
@@ -1,125 +1,125 @@
...
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/gallery_localizations.dart';
+import 'package:gallery/demos/material/material_demo_types.dart';
+
+class DividerDemo extends StatelessWidget {
+  const DividerDemo({super.key, required this.type});
```
Comment (amanv8060) [inline lib/demos/material/divider_demo.dart:10]: CRLF issue, this is the only change in this file. or should I revert the change in this file to try again?

File: lib/demos/material/menu_demo.dart [hunk 3]
```diff
@@ -192,10 +188,7 @@ class _SectionedMenuDemo extends StatelessWidget {
 // a menu whose current value is highlighted and aligned over the
 // list item's center line.
 class _SimpleMenuDemo extends StatefulWidget {
-  const _SimpleMenuDemo({
-    Key? key,
```
Comment (amanv8060) [inline lib/demos/material/menu_demo.dart:196]: this has been removed from private classes which were not providing keys, due to unused element lint. Although they are actually not being used, I am not sure if this is the style.

**Review by amanv8060 — COMMENTED** [review]
@guidezpl can you please update code segments once this is ready to be merged?

File: analysis_options.yaml (hunk 1 above)
Comment (guidezpl) [reply analysis_options.yaml:30]: I think so, since it's not part of `flutter_lints`

File: lib/demos/material/divider_demo.dart (hunk 2 above)
Comment (guidezpl) [reply lib/demos/material/divider_demo.dart:10]: That's fine.

**Review by guidezpl — APPROVED** [review]
Thank for the pull request.

You can update the code segments with 
```
flutter pub get
flutter pub run grinder update-code-segments
```

**Comment by amanv8060** [comment]
Unlike https://github.com/flutter/gallery/pull/608#discussion_r810656533 code viewer worked perfectly fine this time.
